src/hb-ot-layout.cc

```diff
@@ -1009,43 +1027,109 @@ hb_ot_layout_get_size_params (hb_face_t    *face,
-
-      if (params.designSize)
-      {
 #define PARAM(a, A) if (a) *a = params.A
-	PARAM (design_size, designSize);
-	PARAM (subfamily_id, subfamilyID);
-	PARAM (subfamily_name_id, subfamilyNameID);
-	PARAM (range_start, rangeStart);
-	PARAM (range_end, rangeEnd);
+      PARAM (design_size, designSize);
+      PARAM (subfamily_id, subfamilyID);
+      PARAM (subfamily_name_id, subfamilyNameID);
+      PARAM (range_start, rangeStart);
+      PARAM (range_end, rangeEnd);
 #undef PARAM
 
-	return true;
-      }
+      return true;
     }
   }
 
-#define PARAM(a, A) if (a) *a = 0
-  PARAM (design_size, designSize);
-  PARAM (subfamily_id, subfamilyID);
-  PARAM (subfamily_name_id, subfamilyNameID);
-  PARAM (range_start, rangeStart);
-  PARAM (range_end, rangeEnd);
-#undef PARAM
+#define UNPARAM(a) if (a) *a = 0
+  UNPARAM (design_size);
+  UNPARAM (subfamily_id);
+  UNPARAM (subfamily_name_id);
+  UNPARAM (range_start);
+  UNPARAM (range_end);
+#undef UNPARAM
 
   return false;
 }
 
+/**
+ * hb_ot_layout_get_feature_parameters:
+ *
+ * Since: REPLACEME
+ **/
+hb_bool_t
+hb_ot_layout_get_feature_parameters (hb_face_t       *face,
+				     hb_tag_t        tag,
+				     unsigned int    *label_id             /* OUT.  May be nullptr */,
+				     unsigned int    *tooltip_id           /* OUT.  May be nullptr */,
+				     unsigned int    *sample_id            /* OUT.  May be nullptr */,
+				     unsigned int    *num_named_parameters /* OUT.  May be nullptr */,
+				     unsigned int    *first_param_id       /* OUT.  May be nullptr */,
+				     unsigned int    *char_count           /* OUT.  May be nullptr */,
+				     hb_codepoint_t **characters           /* OUT.  May be nullptr */)
+{
+  const OT::FeatureParams &feature_params =
+    _get_gsubgpos_matched_feature_params (face, tag);
+  if (&feature_params != &OT::Null (OT::FeatureParams))
+  {
+    if ((tag & 0xFFFF0000u) == HB_TAG ('s','s','\0','\0')) /* ssXX */
+    {
+      const OT::FeatureParamsStylisticSet& params =
+        feature_params.get_stylistic_set_params (tag);
+
+#define PARAM(a, A) if (a) *a = params.A
+#define UNPARAM(a) if (a) *a = 0
+      PARAM(label_id, uiNameID);
+      // ssXX features don't have the rest
+      UNPARAM(tooltip_id);
+      UNPARAM(sample_id);
+      UNPARAM(num_named_parameters);
+      UNPARAM(first_param_id);
+      UNPARAM(char_count);
+      UNPARAM(characters);
+      return true;
+    }
+    if ((tag & 0xFFFF0000u) == HB_TAG ('c','v','\0','\0')) /* cvXX */
+    {
+      const OT::FeatureParamsCharacterVariants& params =
+        feature_params.get_character_variants_params (tag);
+
+      PARAM(label_id, featUILableNameID);
+      PARAM(tooltip_id, featUITooltipTextNameID);
+      PARAM(sample_id, sampleTextNameID);
+      PARAM(num_named_parameters, numNamedParameters);
+      PARAM(first_param_id, firstParamUILabelNameID);
+      PARAM(char_count, characters.len);
```

Comment:
What if any of these was `NULL` would we just return zero here?

Reply:
I guess no? It follows the idea of `hb_ot_layout_get_size_params`.